TileSystem/Implementation/TwoDimension/Area.cs

```diff
@@ -152,9 +156,29 @@ public ITile Get(IPosition position)
 		/// <returns>List of neighbours or null</returns>
 		public List<ITile> GetNeighbours(ITile tile)
 		{
-			// TODO: Issue 6 (https://github.com/Wizcorp/TileSystem/issues/6)
-			throw new NotImplementedException();
-		}
+            if (tile == null)
+            {
+                throw new ArgumentNullException("tile", "Tile can not be null");
+            }
+
+            if (tiles.Contains(tile) == false)
+            {
+                throw new ArgumentException("Tile not in Area", "tile");
+            }
+
+            List<ITile> neigbourTiles = tiles.FindAll(t => 
```

Comment:
I find it more readable for the delegate passed in to functions like this to be separately declared as an action, however this is quite short. What is your opinion on this?

Reply:
Since the function would need to return a boolean, an action is not possible (Maybe I'm misunderstood on this point).

I came up with this solution : (the indentation is broken here)

`Func<ITile, Predicate<ITile>> ArePositionsNeighbours = lhs => (rhs =>
		Math.Abs(((Position2D)lhs.Position).X - ((Position2D)rhs.Position).X) == 1 ||
		Math.Abs(((Position2D)lhs.Position).Y - ((Position2D)rhs.Position).Y) == 1);`

Which can be used like this : 

`			List<ITile> neigbourTiles = tiles.FindAll(ArePositionsNeighbours(tile));
`

This method uses a Predicate, which is what List.FindAll needs, and allows the code to be separated. However, the code may be even less readable using this solution. On the other hand, using this, we gain re-usability, which in my opinion is really important.

Reply:
Makes sense, in this case knowing it's not going to be reused it's safe to leave it as is.